### PR TITLE
refactor!?: simple forwardMessage

### PR DESCRIPTION
**Pull request by barak007**
This PR changes introduces several changes to improve the communication:

### Fix
- Address the scenario where a window self-registers, preventing potential infinite loops.

### Improve/Fix
- Simplify the message forwarding process by direct forwarding, eliminating the need for a complex intermediary registration and state management.
- Overhaul communication processes to better handle messages, automate environment registration, and improve env disposal mechanisms (don't auto-register an env that is disposing).
- Modify the handling of slowness logs, categorizing them as warnings instead of errors.
- Enhance log messages by incorporating environment information.
- Don't wait for iframe load when starting engine, instead wait for communication ready.
- DEBUG mode for verbose logs (turned on manually for now)

### Chore
- Remove redundant code related to clearing timeouts and callbacks, specifically in `locallyClear` function.
- Clean up unused communication-related code, including `reconnectHandlers, reconnectHandler, and parseHandlerId`.

### CODE REVIEW

File: packages/core/src/com/initializers/iframe.ts [hunk 1]
```diff
@@ -99,8 +105,10 @@ async function startIframe({ com, iframe, instanceId, src, envReadyPromise }: St
         url.searchParams.set(INSTANCE_ID_PARAM_NAME, instanceId);
         iframe.src = url.href;
 
-        await Promise.race([waitForCancel, waitForLoad(iframe), envReadyPromise]);
-
+        await Promise.race([waitForCancel, envReadyPromise]);
```
Comment (idoros) [inline packages/core/src/com/initializers/iframe.ts:108]: check what happens when waitForCancel is called? the registration still happens?

File: packages/core/src/com/communication.ts [hunk 2]
```diff
@@ -82,41 +88,50 @@ export class Communication {
     private disposeListeners = new Set<(envId: string) => void>();
     private callbackToEnvMapping = new Map<string, string>();
     private messageIdPrefix: string;
+    // manual DEBUG_MODE
+    private DEBUG = false;
     constructor(
-        host: Target,
+        private host: Target,
```
Comment (idoros) [inline packages/core/src/com/communication.ts:94]: Its strange to have public/private and non-annotations together? are we sure no one publicly uses the host?

File: packages/core/src/com/communication.ts [hunk 3]
```diff
@@ -439,6 +446,7 @@ export class Communication {
     public envReady(instanceId: string): Promise<void> {
         const { promise, resolve } = deferred();
         if (this.readyEnvs.has(instanceId)) {
+            // the only way an environment can be in readyEnvs is if handleReady was called (why do we need to call it again?)
```
Comment (idoros) [inline packages/core/src/com/communication.ts:449]: we should answer this question

File: packages/core/src/com/communication.ts [hunk 4]
```diff
@@ -472,123 +479,41 @@ export class Communication {
...
             this.sendTo(message.from, {
-                ...responseMessage,
+                from: this.rootEnvId,
+                origin: this.rootEnvId,
+                to: message.origin,
+                callbackId: message.callbackId,
+                type: 'callback',
+                forwardingChain: message.forwardingChain,
                 error: new Error(
                     `cannot reach environment '${message.to}' from '${message.from}' since it's stuck in circular messaging loop`,
```
Comment (idoros) [inline packages/core/src/com/communication.ts:508]: Move to logs

File: packages/core/src/com/communication.ts [hunk 5]
```diff
@@ -675,13 +598,12 @@ export class Communication {
...
                         callbackId,
                         origin,
-                        forwardingChain,
                     };
 
                     this.callWithCallback(envId, message, callbackId, res, rej);
                 }
             } else {
-                throw new Error(`cannot add listener to unconfigured method ${api} ${method}`);
+                throw new Error(`cannot add listener to un-configured method ${api} ${method}`);
```
Comment (idoros) [inline packages/core/src/com/communication.ts:606]: Move to logs

File: packages/core/src/com/communication.ts [hunk 6]
```diff
@@ -985,16 +864,15 @@ export class Communication {
...
 }
 
-const removeMessageArgs = (message: Message): Message => {
-    message = { ...message };
-    if (message.type === 'call' || message.type === 'callback') {
-        const { data } = message;
-        if (data && (data as { args?: unknown }).args) {
-            (data as { args?: unknown }).args = undefined;
-        }
+function cleanMessage(message: Message) {
```
Comment (idoros) [inline packages/core/src/com/communication.ts:867]: clearMessageForLog

**Review by idoros — APPROVED** [review]

File: packages/core/src/com/initializers/iframe.ts [hunk 7]
```diff
@@ -99,8 +105,10 @@ async function startIframe({ com, iframe, instanceId, src, envReadyPromise }: St
...
+        if (!(com as any).environments[instanceId]) {
```
[the remark this answers is not in the document]
Comment (barak007) [reply packages/core/src/com/initializers/iframe.ts:109]: leftovers will be removed

File: packages/core/src/com/initializers/iframe.ts (hunk 1 above)
Comment (barak007) [reply packages/core/src/com/initializers/iframe.ts:108]: not something good. I don't want to change it here. I will change when the refactor of the iframe in codux will be fully resolved

File: packages/core/src/com/communication.ts (hunk 2 above)
Comment (barak007) [reply packages/core/src/com/communication.ts:94]: this turns it into a member. before it wasn't even a member

File: packages/core/src/com/communication.ts (hunk 3 above)
Comment (barak007) [reply packages/core/src/com/communication.ts:449]: we should. but not in this PR. I'm really not sure what's relaying on this behavior.

File: packages/core/src/com/communication.ts (hunk 4 above)
Comment (barak007) [reply packages/core/src/com/communication.ts:508]: removed the log from above and used the proper log message function

File: packages/core/src/com/communication.ts (hunk 5 above)
Comment (barak007) [reply packages/core/src/com/communication.ts:606]: ok. moved one more error.

File: packages/core/src/com/communication.ts (hunk 6 above)
Comment (barak007) [reply packages/core/src/com/communication.ts:867]: done

File: .github/workflows/npm-tag.yml [hunk 8]
```diff
@@ -0,0 +1,23 @@
+name: npm
+on:
+  push:
+    branches:
+      - 'barak/simple-forwarding'
```
[the remark this answers is not in the document]
Comment (barak007) [reply .github/workflows/npm-tag.yml]: removed hook

**Pull request merged**